pkg/cmd/pulumi/crypto.go

```diff
@@ -70,6 +64,8 @@ func getStackSecretsManager(s backend.Stack) (secrets.Manager, error) {
 		switch s.(type) {
 		case filestate.Stack:
 			return newPassphraseSecretsManager(s.Ref().Name(), stackConfigFile)
+		case httpstate.Stack:
```

Comment:
I *think* it's okay (a no-op) to keep the check for `ps.EncryptionSalt != ""` above this case - but the overal structure of this code is a little hard to follow.  There are really three cases here - is it possible to structure this to make the checks for each case clearer?  For example - is the line above this (`return newPassphraseSecretsManager`) even reachable currently?

Reply:
It is reachable when the code has been called as part of creating a new stack - it has no secretsManager at that point and therefore will have an empty secretsProvider AND an empty EncryptionSalt as they have not been added to the config file

Reply:
FWIW, this returns the code in this specific block back to that of what it was before

Reply:
> FWIW, this returns the code in this specific block back to that of what it was before

Okay - great.

> It is reachable when the code has been called as part of creating a new stack - it has no secretsManager at that point and therefore will have an empty secretsProvider AND an empty EncryptionSalt as they have not been added to the config file

I see - definitely would be helpful to note that in a comment here - though recognize this is an issue with the code as it was before you touched it.

Reply:
@lukehoban so we have 2 options here:

1. We work out the test coverage now and risk causing more potential issues for passphrase secrets provider users

2. We ship the patch (as this code is now back to what it was before) and we then work out the testing of the code

Thoughts?

Reply:
Definitely okay with fixing the serious user-facing bug ASAP - but would love to invest in test coverage here as well.